src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -140,16 +145,13 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
             _nuGetProjectContext.OperationId = request.OperationId;
             _isSolutionLoadRestore = isSolutionLoadRestore;
 
-            using (var ctr1 = token.Register(() => _status = NuGetOperationStatus.Cancelled))
+            try
             {
-                try
-                {
-                    await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
-                }
-                catch (OperationCanceledException) when (token.IsCancellationRequested)
-                {
-                }
-                catch (Exception e)
+                await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
+            }
+            catch (Exception e)
+            {
+                if (!token.IsCancellationRequested)
```

Comment:
Same here. Prefer catching the actual exception and then checking the token.

Reply:
code changes to handle `OperationCancelledException`